presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryClient.java

```diff
@@ -0,0 +1,182 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.bigquery;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.DatasetId;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableDefinition;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import com.google.cloud.bigquery.TableResult;
+import com.google.cloud.http.BaseHttpServiceException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterators;
+
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.StreamSupport;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.UUID.randomUUID;
+import static java.util.stream.Collectors.joining;
+
+// holds caches and mappings
+// presto converts the dataset and table names to lower case, while BigQuery is case sensitive
+// the mappings here keep the mappings
+public class BigQueryClient
+{
+    private final BigQuery bigQuery;
+    private final Optional<String> viewMaterializationProject;
+    private final Optional<String> viewMaterializationDataset;
+    private final ConcurrentMap<TableId, TableId> tableIds = new ConcurrentHashMap<>();
+    private final ConcurrentMap<DatasetId, DatasetId> datasetIds = new ConcurrentHashMap<>();
+
+    BigQueryClient(BigQuery bigQuery, BigQueryConfig config)
+    {
+        this.bigQuery = bigQuery;
+        this.viewMaterializationProject = config.getViewMaterializationProject();
+        this.viewMaterializationDataset = config.getViewMaterializationDataset();
+    }
+
+    // return empty if no filters are used
+    private static Optional<String> createWhereClause(String[] filters)
+    {
+        return Optional.empty();
+    }
+
+    TableInfo getTable(TableId tableId)
+    {
+        TableId bigQueryTableId = tableIds.get(tableId);
+        Table table = bigQuery.getTable(bigQueryTableId != null ? bigQueryTableId : tableId);
+        if (table != null) {
+            tableIds.putIfAbsent(tableId, table.getTableId());
+            datasetIds.putIfAbsent(toDatasetId(tableId), toDatasetId(table.getTableId()));
+        }
+        return table;
+    }
+
+    DatasetId toDatasetId(TableId tableId)
```

Comment:
shall we take a review of all access modifiers? we are using default everywhere

Reply:
yes fixed, most methods should be protected within the module's scope